protocol.md

```diff
@@ -726,6 +727,141 @@ Upload-Length: 11
 Upload-Concat: final;/files/a /files/b
 ```
 
+### Challenge
+
+With this extension, Clients can authenticate follow-up requests to a [Creation](#creation) or [Creation With Upload](#creation-with-upload) 
+`POST` request using a high-entropy cryptographic random shared secret - and challenges
+based thereon.
+
+If the Server supports this extension, it MUST add `challenge` to the `Tus-Extension` header.
+
+#### Headers
+
+##### Upload-Secret
+
+The `Upload-Secret` MUST be a high-entropy cryptographic random string, consisting
+of 48 to 256 characters from the [base64 alphabet](https://tools.ietf.org/html/rfc4648#section-4).
+That's 36 to 192 bytes encoded as base64.
+
+The `Upload-Secret` header MUST NOT be included with anything but `POST` requests
+that create a new upload resource. Servers receiving an `Upload-Secret` header with
+any other request MUST respond with a `400 Bad Request` status.
+
+##### Upload-Challenge
+
+If an `Upload-Secret` header was sent with the `POST` request that created
+the upload resource, Clients MUST include an `Upload-Challenge` header with every
+subsequent request targeting that upload resource.
+
+The value of `Upload-Challenge` is defined as follows:
+
+```
+Upload-Challenge = SHA256([Upload-Secret] + SHA256(`[HTTP Method]` + [Upload-Offset] + [Upload-Secret] + [Content-Length]))
+```
+
+The *effective* HTTP method of the request that the `Upload-Challenge` is sent with MUST be
+used in place of `HTTP Method`. If `X-HTTP-Method-Override` is used in the same request,
+`HTTP method` needs to be replaced with the value of `X-HTTP-Method-Override`.
+
+Where `Upload-Offset` and `Content-Length` headers are used in the same request, their
+values MUST be used as the respective `Upload-Offset` and `Content-Length` values. A
+value of `0` MUST be used for `Upload-Offset` and `Content-Length` where no headers
+of the same name are sent with the request.
```

Comment:
Could we clarify that "Upload-Length" here is the length of the body being sent? It was not clear to me at first and I read it as the "Upload-Length" specified when creating the file.

Reply:
I'm struggling to find good words to add such a clarification. Any suggestion?

Reply:
@smatsson Do you mean the Content-Length header here and not the the Upload-Length header?

Reply:
@Acconut Yes, my bad! Sorry about that.

Reply:
Closing this as resolved since I think we pretty much agree `Content-Length` will be removed from the equation.